flash/core/model.py

```diff
@@ -163,22 +171,36 @@ def configure_optimizers(self) -> torch.optim.Optimizer:
     def data_pipeline(self) -> DataPipeline:
         # we need to save the pipeline in case this class
         # is loaded from checkpoint and used to predict
-        if not self._data_pipeline:
-            try:
-                # datamodule pipeline takes priority
-                self._data_pipeline = self.trainer.datamodule.data_pipeline
-            except AttributeError:
-                self._data_pipeline = self.default_pipeline()
-        return self._data_pipeline
+        return self._get_pipeline('data')
 
     @data_pipeline.setter
     def data_pipeline(self, data_pipeline: DataPipeline) -> None:
         self._data_pipeline = data_pipeline
 
+    @property
+    def postprocessing_pipeline(self) -> PostProcessingPipeline:
+        return self._get_pipeline('postprocessing')
+
+    def _get_pipeline(self, pipeline_type: str):
+        pipeline_attr_name = f'{pipeline_type}_pipline'
+
+        if getattr(self, '_' + pipeline_attr_name) is not None:
+            return getattr(self, '_' + pipeline_attr_name)
+
+        if self.datamodule is not None and hasattr(self, pipeline_attr_name):
+            return getattr(self.datamodule, pipeline_attr_name)
+
+        if self.trainer is not None and hasattr(self.trainer, 'datamodule') and self.trainer.datamodule is not None:
+            if hasattr(self.trainer.datamodule,
+                       pipeline_attr_name) and getattr(self.trainer.datamodule, pipeline_attr_name is not None):
```

Comment:
When can `pipeline_attr_name ` be None ?

Reply:
It can't, that should be outside the brackets :)